app/services/trainees/create_from_dttp.rb

```diff
@@ -0,0 +1,333 @@
+        ),
+      ]
+    end
+
+    def disability_attributes
+      if disability.blank?
+        return {
+          disability_disclosure: Diversities::DISABILITY_DISCLOSURE_ENUMS[:not_provided],
+        }
+      end
+
+      if disability == Diversities::NO_KNOWN_DISABILITY
+        return {
+          diversity_disclosure: Diversities::DIVERSITY_DISCLOSURE_ENUMS[:diversity_disclosed],
+          disability_disclosure: Diversities::DISABILITY_DISCLOSURE_ENUMS[:no_disability],
+        }
+      end
+
+      # TODO: This needs a decision, since DTTP may have 'multiple disabilities'
+      if disability == Diversities::MULTIPLE_DISABILITIES
+        return {
+          diversity_disclosure: Diversities::DIVERSITY_DISCLOSURE_ENUMS[:diversity_disclosed],
+          disability_disclosure: Diversities::DISABILITY_DISCLOSURE_ENUMS[:disabled],
+          disabilities: Disability.where(name: ::Diversities::OTHER),
+        }
+      end
+
+      {
+        diversity_disclosure: Diversities::DIVERSITY_DISCLOSURE_ENUMS[:diversity_disclosed],
+        disability_disclosure: Diversities::DISABILITY_DISCLOSURE_ENUMS[:disabled],
+        disabilities: Disability.where(name: disability),
+      }
+    end
+
+    def disability
+      @disability ||= find_by_entity_id(
+        dttp_trainee.response["_dfe_disibilityid_value"],
+        Dttp::CodeSets::Disabilities::MAPPING,
+      )
+    end
+
+    def ethnicity_attributes
+      ethnic_group = Diversities::BACKGROUNDS.select { |_key, values| values.include?(ethnic_background) }&.keys&.first
+
+      if ethnic_background.present? && ethnic_background != Diversities::NOT_PROVIDED
+        diversity_disclosure = Diversities::DIVERSITY_DISCLOSURE_ENUMS[:diversity_disclosed]
+      else
+        diversity_disclosure = Diversities::DIVERSITY_DISCLOSURE_ENUMS[:diversity_not_disclosed]
+      end
+
+      if Diversities::BACKGROUNDS.values.flatten.include?(ethnic_background)
+        return {
+          ethnic_background: ethnic_background,
+          ethnic_group: ethnic_group,
+          diversity_disclosure: diversity_disclosure,
+        }
+      end
+
+      {}
+    end
+
+    def ethnic_background
+      @ethnic_background ||= find_by_entity_id(
+        dttp_trainee.response["_dfe_ethnicityid_value"],
+        Dttp::CodeSets::Ethnicities::MAPPING,
+      )
+    end
+
+    def ethnicity_and_disability_attributes
+      ethnicity_attributes.merge(disability_attributes)
+    end
+
+    def personal_details_attributes
+      {
+        first_names: dttp_trainee.response["firstname"],
+        middle_names: dttp_trainee.response["middlename"],
+        last_name: dttp_trainee.response["lastname"],
+        date_of_birth: dttp_trainee.date_of_birth,
+        gender: trainee_gender,
+        nationalities: nationalities,
+      }
+    end
+
+    def locale_code
+      return {} if dttp_trainee.response["address1_line1"].blank?
+
+      { locale_code: Trainee.locale_codes[:uk] }
```

Comment:
Is it ok to assume this?

Reply:
I think so as international addresses don't use the `address1_line1` field, but not a 100% certain. (From my understanding we don't send international addresses to dttp at all)
https://github.com/DFE-Digital/register-trainee-teachers/blob/e5a50a747ef7f04b6db33919d09f1ff3c4c28b9c/app/lib/dttp/params/contact.rb#L39-L42